lib/graphql/graphql-token-fee-fetcher.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  ITokenFeeFetcher,
+  OnChainTokenFeeFetcher,
+} from '@uniswap/smart-order-router/build/main/providers/token-fee-fetcher'
+import { IUniGraphQLProvider, UniGraphQLProvider } from './graphql-provider'
+import { TokenFeeMap } from '@uniswap/smart-order-router/build/main/providers/token-fee-fetcher'
+import { ProviderConfig } from '@uniswap/smart-order-router/build/main/providers/provider'
+import { TokensInfoResponse } from './graphql-schemas'
+import { BigNumber } from 'ethers'
+import { BaseProvider } from '@ethersproject/providers'
+import { ChainId } from '@uniswap/sdk-core'
+import { metric } from '@uniswap/smart-order-router/build/main/util/metric'
+import { log, MetricLoggerUnit } from '@uniswap/smart-order-router'
+
+/* Implementation of the ITokenFeeFetcher interface to give access to Uniswap GraphQL API token fee data.
+ * This fetcher is used to get token fees from GraphQL API and fallback to OnChainTokenFeeFetcher if GraphQL API fails
+ * or not all addresses could be fetched.
+ * Note: OnChainTokenFeeFetcher takes into account the provided blocknumber when retrieving token fees (through providerConfig),
+ * but GraphQLTokenFeeFetcher always returns the latest token fee (GraphQl doesn't keep historical data).
+ * FOT tax doesn't change often, hence ok to not use blocknumber here.
+ * */
+export class GraphQLTokenFeeFetcher implements ITokenFeeFetcher {
+  private readonly graphQLProvider: IUniGraphQLProvider
+  private readonly onChainFeeFetcherFallback: ITokenFeeFetcher
+  private readonly chainId: ChainId
+
+  constructor(chainId: ChainId, rpcProvider: BaseProvider) {
+    this.graphQLProvider = new UniGraphQLProvider()
+    this.onChainFeeFetcherFallback = new OnChainTokenFeeFetcher(chainId, rpcProvider)
+    this.chainId = chainId
+  }
```

Comment:
I'd suggest we pass the OnChainTokenFeeFetcher as a constructor argument, rather than initialized here, probably the same with the graphql provider, to follow Inversion of Control patterns